packages/shared/src/utils/contract/contract.ts

```diff
@@ -34,12 +55,44 @@ export const hasContractEntered = (contract_info: TContractInfo) => !!contract_i
 
 export const isAccumulatorContract = (contract_type: string) => /ACCU/i.test(contract_type);
 
+export const isAccumulatorContractOpen = (
+    {
+        contract_type,
+        current_spot,
+        exit_tick,
+        high_barrier,
+        low_barrier,
+        status,
+    }: TContractInfo | TContractInfoWithNumericBarriers = {},
+    in_pixels?: boolean
+) => {
+    const has_crossed_barriers =
+        !!(current_spot && high_barrier && low_barrier) &&
+        (in_pixels
+            ? current_spot <= +high_barrier || current_spot >= +low_barrier
+            : current_spot >= +high_barrier || current_spot <= +low_barrier);
+    return isAccumulatorContract(contract_type || '') && status === 'open' && !has_crossed_barriers && !exit_tick;
+};
+
 export const isMultiplierContract = (contract_type: string) => /MULT/i.test(contract_type);
 
 export const isVanillaContract = (contract_type: string) => /VANILLA/i.test(contract_type);
 
 export const isCryptoContract = (underlying: string) => /^cry/.test(underlying);
 
+export const getAccuBarriersDelayTimeMs = (symbol: string) => {
+    return symbols_1s.includes(symbol) ? DELAY_TIME_1S_SYMBOL : DELAY_TIME_1S_SYMBOL * 2;
+};
+
+export const getAccuBarriersForContractDetails = (contract_info: TContractInfo) => {
+    if (!isAccumulatorContract(contract_info.contract_type || '')) return {};
+    const is_accu_contract_open = isAccumulatorContractOpen(contract_info);
+    const { current_spot_high_barrier, current_spot_low_barrier, high_barrier, low_barrier } = contract_info || {};
+    const accu_high_barrier = is_accu_contract_open ? current_spot_high_barrier : high_barrier;
+    const accu_low_barrier = is_accu_contract_open ? current_spot_low_barrier : low_barrier;
+    return { accu_high_barrier, accu_low_barrier };
```

Comment:
```suggestion
    return { accu_high_barrier, accu_low_barrier };
```
We can coerce to number here instead of doing it on where we use this function.

Reply:
@yashim-deriv the thing is that in most places, we need the barriers to be returned as `string`s, and there's only 1 place (contract-store L349) where we need them to be `numbers`.